src/toil/cwl/cwltoil.py

```diff
@@ -987,6 +986,13 @@ def visitSteps(t, op):
             visitSteps(s.embedded_tool, op)
 
 
+def cleanTempDirs(job):
+    """Remove temporarly created directories."""
+    if job._succeeded:
+        for tempDir in job.openTempDirs:
+            shutil.rmtree(tempDir)
```

Comment:
You may wish to have an `if os.path.exists():` statement here, as well as resetting with: `job.openTempDirs = []` when it has finished deleting the directories, just to be safe.

Reply:
Good call.